src/components/Home.test.js

```diff
@@ -1,17 +1,78 @@
 import React from 'react';
 import { shallow } from 'enzyme';
-import Home from './Home';
+import { HomeComponent, HomeButtonsMajor, HomeButtonsMinor } from './Home';
+import { Link } from 'react-router-dom';
+Link.displayName = 'Link';
+jest.mock('@material-ui/core/Button', () => 'Button');
 jest.mock('@material-ui/core/Grid', () => 'Grid');
+jest.mock('@material-ui/core/IconButton', () => 'IconButton');
+jest.mock('@material-ui/icons/Call', () => 'CallIcon');
+jest.mock('@material-ui/icons/Restaurant', () => 'RestaurantIcon');
+jest.mock('@material-ui/icons/Home', () => 'HomeIcon');
+jest.mock('@material-ui/icons/HotTub', () => 'HotTubIcon');
+jest.mock('@material-ui/icons/DirectionsBus', () => 'BusIcon');
+jest.mock('@material-ui/icons/LocalGroceryStore', () => 'GroceryStoreIcon');
+jest.mock('@material-ui/icons/Healing', () => 'HealingIcon');
+jest.mock('@material-ui/icons/LibraryBooks', () => 'LibraryBooksIcon');
+jest.mock('@material-ui/icons/Pets', () => 'PetsIcon');
+jest.mock('@material-ui/icons/Wifi', () => 'WifiIcon');
+jest.mock('@material-ui/icons/Work', () => 'WorkIcon');
+jest.mock('@material-ui/icons/Info', () => 'InfoIcon');
 jest.mock('./Search', () => 'Search');
 
 describe('<Home/>', () => {
-  const wrapper = shallow(<Home />);
+  const wrapper = shallow(<HomeComponent />);
+
+  const numSearchGridComponents = 3;
+  const numButtonsMajorGridComponents = 2 + HomeButtonsMajor.length * 2;
+  const numButtonsMinorGridComponents = 2 + HomeButtonsMinor.length * 2;
   it('renders the proper number of layout components', () => {
-    expect(wrapper.find('Grid').length).toBe(2);
+    expect(wrapper.find('Grid').length).toBe(
+      1 +
+        numSearchGridComponents +
+        numButtonsMajorGridComponents +
+        numButtonsMinorGridComponents
+    );
   });
   it('renders one layout component as the search component', () => {
     expect(
       wrapper.findWhere(n => n.prop('component') === 'Search').length
     ).toBe(1);
   });
+  it('renders all major buttons and only once', () => {
+    HomeButtonsMajor.forEach(button => {
+      let matchingButtons = wrapper
```

Comment:
rather pedantic but, if `matchingButtons` is never reassigned use `const` instead of `let`

Reply:
That's in a forEach and reassigned every loop, but it would make sense to define the variable outside of the loop.

Reply:
a new context gets created with each function call, so you can use `const`

Reply:
```suggestion
      const matchingButtons = wrapper;
```